client/src/rpc_client.rs

```diff
@@ -1146,7 +1147,14 @@ impl RpcClient {
                 signature,
             ));
             sleep(Duration::from_millis(500));
-            confirmations = self.get_num_blocks_since_signature_confirmation(&signature)?;
+            confirmations = self
+                .get_num_blocks_since_signature_confirmation(&signature)
+                .unwrap_or(confirmations);
+            if now.elapsed().as_secs() >= 120 {
+                return Err(
+                    RpcError::ForUser("transaction not finalized. This can happen when a transaction lands in a minor fork. Please retry.".to_string()).into(),
```

Comment:
wdyt about abandoned instead of minor?
```suggestion
                    RpcError::ForUser("transaction not finalized. This can happen when a transaction lands in an abandoned fork. Please retry.".to_string()).into(),
```